### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by HarshithaKonduru214**
Hi, pls review.
chrome-capture-2022-2-27
chrome-capture-2022-2-27 (1)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import "./App.css";
+import { Routes, Route } from "react-router-dom"
+import { Home } from "./pages/Home/Home"
+import { Product } from "./pages/Product-Listing/Product"
```
Comment (rohanmathur91) [inline src/App.js]: try barrel exports

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { MockAPI } from "./components/index";
```
Comment (rohanmathur91) [inline src/App.js]: No need to mentions /index

File: src/Context/ProductContext.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { useContext, createContext, useReducer } from "react";
+import { brands } from "../data"
+import { productReducer } from "../Reducer/ProductReducer";
+import { sortData } from "../utils/sortData";
+import { filterGender } from "../utils/filterGender"
+import { filterInStock } from "../utils/filterInStock"
+import { filteredBrands } from "../utils/filterBrands";
+import { filterPrice } from "../utils/filterPrice";
```
Comment (rohanmathur91) [inline src/Context/ProductContext.js]: suggestion: try barrel exports, the way you have done for components

File: src/components/Nav.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import "./style.css";
+
+import { Link } from "react-router-dom";
+import { FaHeart, FaUser, FaShoppingCart } from "react-icons/fa"
+import { brands, types } from "../data"
+
+const Nav = () => {
+    
+    return (
+        <nav className="nav mb-2 box-shadow">
```
Comment (rohanmathur91) [inline src/components/Nav.jsx]: take care of spacing

File: src/components/ShopBy.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import { Link } from "react-router-dom"
+import { brands } from "../data"
+import axios from "axios";
+import "./style.css";
+
+export const ShopBy = () => {
+	const [categoryList, setCategorList] = useState([]);
+    const price = [1000, 2000, 3000, 4000];
```
Comment (rohanmathur91) [inline src/components/ShopBy.jsx:9]: indent the code properly

File: src/components/Sidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                    <input className="checkbox mr-1" type="checkbox" name="gucci" id="gucci" onChange={(e) => dispatch({type:"FILTER_BY_BRAND", payload: "gucci"})}/>Gucci</label>
+                    <br />
+                    <label htmlFor="nykaa">
+                    <input className="checkbox mr-1" type="checkbox" name="nykaa" id="nykaa" onChange={(e) => dispatch({type:"FILTER_BY_BRAND", payload: "nykaa"})}/>Nykaa</label>
+                    <br />
+                    <label htmlFor="engage">
+                    <input className="checkbox mr-1" type="checkbox" name="engage" id="engage" onChange={(e) => dispatch({type:"FILTER_BY_BRAND", payload: "engage"})}/>Engage</label>
+                    <br />
+                    <label htmlFor="out-of-stock">
+                    <input className="checkbox mr-1" type="checkbox" name="out-of-stock" id="out-of-stock" onChange={(e) => dispatch({type:"OUT_OF_STOCK", payload: "out-of-stock"})}/>Include Out of Stock</label>
```
Comment (rohanmathur91) [inline src/components/Sidebar.jsx:66]: Suggestion:  try to have an array of data for your filters and just loop over them

File: src/pages/Home/Home.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                <div>
+                    <Footer />
+                </div>
+
+                
+                
+                
+            </main>
+            
+        </div>
```
Comment (rohanmathur91) [inline src/pages/Home/Home.jsx]: remove unnecessary space

**Review by rohanmathur91 — APPROVED** [review]
Added few suggestions
- Liked the way you have handled different icons using switch statements (I learned something new)
- Good breakdown of components
- File structuring is also good, just try to export the files the way you have done for components (ie barrel exports) wherever possible

**Review by anupkgurung — APPROVED** [review]
good to see switch case instead of if/else.
suggestion - try to keep the merge clean(exclude non required files)

File: src/components/Sidebar.jsx (hunk 6 above)
Comment (HarshithaKonduru214) [reply src/components/Sidebar.jsx:66]: Not sure why it didnt work as expected, would resolve it soon. Thanks !

**Pull request merged**